### PR TITLE
Enhance funnel chart label display policy

### PR DESCRIPTION
**Pull request by cuijian-dexter**
1. The value of the fan page of the pie chart is less than a certain value.  The label and labelLine of this fan page are not displayed.
2. Funnel plot position increase insideLeft and insideRight attributes.

Add the limitScale parameter under series. Label, and the parameter value is greater than 0 and less than 1.
To ensure that the pie chart does not show label and labelLine on the fan page that is less than a certain value.
```js
options={
  title: {
    text: '饼图标题',
    left: 16,
    textStyle: {
      fontSize: 14
    }
  },
  tooltip: {
    trigger: 'item',
    axisPointer: {
      //坐标轴指示器，坐标轴触发有效type: 'shadow'//默认为直线，可选为：'line'|'shadow'
    },
    extraCssText: 'max-width: 300px; word-break:break-all; white-space: normal'
  },
  legend: {
    data: [
      
    ]
  },
  series: [
    {
      type: 'pie',
      label: {
        normal: {
          show: true,
          limitScale: 0.02//小于2%的，不显示
        },
        emphasis: {
          show: true,
          textStyle: {
            fontSize: '14',
            fontWeight: 'bold'
          }
        }
      },
      labelLine: {
        lineStyle: {
          color: '#D9D9D9'
        }
      },
      data: [
        
      ]
    }
  ]
};
```
image
image

### CODE REVIEW

**Comment by cuijian-dexter** [comment]
Duplicate of #8390 
Duplicate of #8507

**Comment by Ovilia** [comment]
@cuijian-dexter Hi @cuijian-dexter , I think this feature is very useful although we have to discuss it in details. But first of all, could you please change the title and comments into English so that more community members can understand? Thanks!

**Comment by cuijian-dexter** [comment]
@Ovilia  ok,I'll change it right away

File: src/chart/funnel/funnelLayout.js [hunk 1]
```diff
@@ -71,9 +71,19 @@ function labelLayout(data) {
         var linePoints;
 
         if (isLabelInside) {
-            textX = (points[0][0] + points[1][0] + points[2][0] + points[3][0]) / 4;
-            textY = (points[0][1] + points[1][1] + points[2][1] + points[3][1]) / 4;
-            textAlign = 'center';
+            if (labelPosition === 'insideLeft'){
```
Comment (Ovilia) [inline src/chart/funnel/funnelLayout.js]: Add space: `if (labelPosition === 'insideLeft') {`.
Also change the similar problems.

File: src/chart/funnel/funnelLayout.js [hunk 2]
```diff
@@ -71,9 +71,19 @@ function labelLayout(data) {
...
+                textX = (points[0][0] + points[3][0])/2 + 5;
```
Comment (Ovilia) [inline src/chart/funnel/funnelLayout.js]: `) / 2`

**Comment by Ovilia** [comment]
@cuijian-dexter The pie chart label filtering and funnel label positioning seems to be two different topics. I reviewed the funnel part and that seems to be working. How about removing the pie chart part in this PR (you can make another PR later)?
And please format your code as I mentioned above.
Thanks for your contribution.

**Comment by Ovilia** [comment]
You need to remove the pie-related code from the PR. Undo the changes in pie chart; commit; pull request.
Here's how to open another PR on GitHub: https://stackoverflow.com/questions/8450036/how-to-open-multiple-pull-requests-on-github You can also wait for the PR to be closed and then add a new PR.

**Comment by cuijian-dexter** [comment]
@Ovilia Okay, I have deleted the pie-related code.

**Comment by Ovilia** [comment]
@cuijian-dexter Very close now! Please see https://github.com/apache/incubator-echarts/pull/8759/files for my comments and fix them. And revert the changes in `src/chart/pie/pieLayout.js`.

**Comment by cuijian-dexter** [comment]
@Ovilia should I  delete this file src/chart/pie/pieLayout.js ?

**Comment by cuijian-dexter** [comment]
@Ovilia It seems to be OK. I'm very sorry, this is my first time to contribute code, not very skilled. 
Thank you for your help. I will try my best.

**Comment by Ovilia** [comment]
Please don't be sorry. That's OK and we very appreciate your contribution.
One problem here. You need to fix the code style as I commented at https://github.com/apache/incubator-echarts/pull/8759/files .

**Comment by cuijian-dexter** [comment]
@Ovilia ok, Already fixed the code style.Add space

File: src/chart/funnel/funnelLayout.js [hunk 3]
```diff
@@ -71,9 +71,19 @@ function labelLayout(data) {
...
 
         if (isLabelInside) {
-            textX = (points[0][0] + points[1][0] + points[2][0] + points[3][0]) / 4;
-            textY = (points[0][1] + points[1][1] + points[2][1] + points[3][1]) / 4;
-            textAlign = 'center';
+            if (labelPosition === 'insideLeft') {
+                textX = (points[0][0] + points[3][0]) / 2 + 5;
+                textY = (points[0][1] + points[3][1]) / 2;
+                textAlign = 'left';
+            }else if (labelPosition === 'insideRight') {
```
Comment (Ovilia) [inline src/chart/funnel/funnelLayout.js]: It should be `} else if`

File: src/chart/funnel/funnelLayout.js [hunk 4]
```diff
@@ -71,9 +71,19 @@ function labelLayout(data) {
...
+                textX = (points[1][0] + points[2][0]) / 2 - 5;
+                textY = (points[1][1] + points[2][1]) / 2;
+                textAlign = 'right';
+            }else {
```
Comment (Ovilia) [inline src/chart/funnel/funnelLayout.js]: `} else {`

**Comment by cuijian-dexter** [comment]
@Ovilia Have already been solved

**Comment by Ovilia** [comment]
@cuijian-dexter Thanks for your contribution! 🌟🌟🌟🌟🌟 
Now you are one of the ECharts contributors.

And are you interested in being an ECharts **committer**? This has more restrict rules than contributors. Committers are those who contribute more frequently and play more important roles on ECharts project.

After being a committer:
- You can have write access to the ECharts code, so that PR is not needed any more
- You can mange issues (add tags, assign to someone, ...)
- You can discuss about features with us committers more frequently with Email
- You name and personal page link will be displayed at [ECharts Website About page](http://echartsjs.com/about.html)
- See [this](https://www.apache.org/dev/new-committers-guide.html) for more information

If you are interested, here's what you can do:
- Subscribe our mailing list by send an email to dev-subscribe@echarts.apache.org
- When you want to discuss with us about something, send email to dev@echarts.apache.org
- Help others by replying issues regularly
- Make more useful pull requests (fix bugs or make enhancement in issues)

Here are some issues that may be easier to be fixed: https://github.com/apache/incubator-echarts/projects/3

And you will be invited to be a committer when we think your contribution and knowledge of ECharts project is enough. We are looking forward to this! 😃

**Comment by cuijian-dexter** [comment]
@Ovilia Thank you
I am very happy to be one of the contributors to the project.
I am interested in becoming a committer.
Whenever I have time, I will try my best to help solve the problem. I hope to be a committer early.

**Comment by Ovilia** [comment]
@cuijian-dexter Thanks, please send scanned PDF as soon as possible. ⭐️

**Comment by cuijian-dexter** [comment]
@Ovilia It has been sent  the scanned file to dev@echarts.apache.org.Please check it.

**Comment by Ovilia** [comment]
@cuijian-dexter Sorry for the late reply. I didn't receive the file from dev@echarts.apache.org. 
Did you subscribed the mailing list? Could you receive it?
Could you please send it to oviliazhang@gmail.com?

**Comment by cuijian-dexter** [comment]
@Ovilia  ok

**Comment by Ovilia** [comment]
I've send the email but the secretary did not reply. There are some other new contributors sending ICLA file and got no reply. I'll send an email to mentor about this.

**Comment by cuijian-dexter** [comment]
@Ovilia ok

**Comment by Ovilia** [comment]
I got an email that your ICLA file has been accepted. Congratulations!

**Comment by cuijian-dexter** [comment]
@Ovilia Thank you!😄

**Pull request merged**